pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -516,7 +533,7 @@ func (r *ReconcileBareMetalHost) actionRegistering(prov provisioner.Provisioner,
 func (r *ReconcileBareMetalHost) actionInspecting(prov provisioner.Provisioner, info *reconcileInfo) actionResult {
 	info.log.Info("inspecting hardware")
 
-	provResult, details, err := prov.InspectHardware()
+	provResult, details, err := prov.InspectHardware(hasInspectAnnotation(info.host))
```

Comment:
I don't love that we're relying on the annotation still being present to control behaviour (i.e. whether we always start a new inspection or accept any existing data that happens to still be in the ironic database).